### ISSUE TITLE
Allow disable of SwitchCase indent rule

### ISSUE THREAD

**Issue by jez9999** [eslint/eslint#7270]
I like my switch case statements to be indented like so:

```
switch (foo) {
    case "bar":
    //...
    break;

    case "xyz":
    //...
    break;

    default:
    //....
}
```

As the current indent rules don't allow enforcing this, perhaps it would be a good option to allow {"SwitchCase": "off"} to be an option for the "indent" rule config.

More info:  
- What rule do you want to change?
  indent
- Does this change cause the rule to produce more or fewer warnings?
  Fewer
- How will the change be implemented? (New option, new default behavior, etc.)?
  New value for SwitchCase option.
- Please provide some example code that this change will affect:

```
switch (foo) {
    case "bar":
    //...
    break;

    case "xyz":
    //...
    break;

    default:
    //....
}
```
- What does the rule currently do for this code?
  Produces error/warning about 'case' indentation.
- What will the rule do after it's changed?
  Produce no error/warning.

**Comment by eslintbot** [comment]
Hi @jez9999, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Hi @jez9999, we can certainly consider a rule enhancement request for this use case. That said, could you please follow the link provided by eslintbot above and edit your post per the instructions on that page? It really helps us to have all the information filled out in your first post.

I also wanted to note that as a workaround, you could disable the indent rule within the switch statement:

```js
switch (foo) {
    /* eslint-disable indent */
    case "bar":
    //...
    break;

    case "xyz":
    //...
    break;

    default:
    //....
    /* eslint-enable indent */
}
```

**Comment by jez9999** [comment]
I have updated the original post with more information.

**Comment by platinumazure** [comment]
Thanks @jez9999! We'll evaluate in the next couple of days.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
